bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -186,6 +193,83 @@ private String getTransactionLogDirectory() {
+            return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+        } else {
+            int transactionAutoCommitTimeout;
+            if (value instanceof Long) {
+                transactionAutoCommitTimeout = ((Long) value).intValue();
+            } else if (value instanceof Integer) {
+                transactionAutoCommitTimeout = (Integer) value;
+            } else {
+                diagnosticLog.warn(ErrorCodes.INVALID_TRANSACTION_AUTO_COMMIT_TIMEOUT, null, value,
+                        DEFAULT_TRX_AUTO_COMMIT_TIMEOUT);
+                if (!diagnosticLog.getDiagnosticList().isEmpty()) {
+                    RuntimeUtils.handleDiagnosticErrors(diagnosticLog);
+                }
+                return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+            }
+            if (transactionAutoCommitTimeout <= 0) {
+                diagnosticLog.warn(ErrorCodes.INVALID_TRANSACTION_CLEANUP_TIMEOUT, null,
+                        transactionAutoCommitTimeout, DEFAULT_TRX_AUTO_COMMIT_TIMEOUT);
+                if (!diagnosticLog.getDiagnosticList().isEmpty()) {
+                    RuntimeUtils.handleDiagnosticErrors(diagnosticLog);
+                }
+                return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+            } else {
+                return transactionAutoCommitTimeout;
+            }
+        }
+    }
+
+    /**
+     * This method gets the user specified config for cleaning up dead transactions. Default is 600.
+     *
+     * @return int transaction cleanup after value
+     */
+    public int getTransactionCleanupTimeout() {
+        VariableKey transactionCleanupTimeoutKey = new VariableKey(TRANSACTION_PACKAGE_ID,
+                TRANSACTION_CLEANUP_TIMEOUT_KEY,
+                PredefinedTypes.TYPE_INT, false);
+        Object value = ConfigMap.get(transactionCleanupTimeoutKey);
+        if (value == null) {
+            return DEFAULT_TRX_CLEANUP_TIMEOUT;
+        } else {
+            int transactionCleanupTimeout;
+            if (value instanceof Long) {
+                transactionCleanupTimeout = ((Long) value).intValue();
+            } else if (value instanceof Integer) {
+                transactionCleanupTimeout = (Integer) value;
+            } else {
+                diagnosticLog.warn(ErrorCodes.INVALID_TRANSACTION_CLEANUP_TIMEOUT, null, value,
+                        DEFAULT_TRX_CLEANUP_TIMEOUT);
+                if (!diagnosticLog.getDiagnosticList().isEmpty()) {
+                    RuntimeUtils.handleDiagnosticErrors(diagnosticLog);
+                }
+                return DEFAULT_TRX_CLEANUP_TIMEOUT;
+            }
+            if (transactionCleanupTimeout <= 0) {
+                diagnosticLog.warn(ErrorCodes.INVALID_TRANSACTION_CLEANUP_TIMEOUT, null,
+                        transactionCleanupTimeout, DEFAULT_TRX_CLEANUP_TIMEOUT);
+                if (!diagnosticLog.getDiagnosticList().isEmpty()) {
+                    RuntimeUtils.handleDiagnosticErrors(diagnosticLog);
+                }
```

Comment:
```suggestion
                diagnosticLog.warn(ErrorCodes.INVALID_TRANSACTION_CLEANUP_TIMEOUT, null,
                        transactionCleanupTimeout, DEFAULT_TRX_CLEANUP_TIMEOUT);
                        RuntimeUtils.handleDiagnosticErrors(diagnosticLog);
```
No need to check `diagnosticLog.getDiagnosticList().isEmpty()`